controlplane/internal/dal/postgres.go

```diff
@@ -458,6 +458,39 @@ func (d *Postgres) InsertMetricEntry(ctx context.Context, metric Metric) error {
 	})))
 }
 
+func (d *Postgres) GetMetricsForSourceModules(ctx context.Context, modules []string) ([]Metric, error) {
+	dbMetrics, err := d.db.GetMetricsBySourceModules(ctx, modules)
+	if err != nil {
+		return nil, errors.Wrap(translatePGError(err), "could not get metrics")
+	}
+	metrics, err := slices.MapErr(dbMetrics, func(in sql.Metric) (Metric, error) {
+		var datapoint DataPoint
+		if err := json.Unmarshal((in.Value), &datapoint); err != nil {
+			return Metric{}, errors.Wrapf(err, "could not unmarshal datapoint for row %q: ", in.ID)
```

Comment:
Does `%q` work with numbers?

Reply:
Ha! Good catch. No idea, but I'm guessing it should be %d here instead :). Updated.